crates/oxc_cli/src/git.rs

```diff
@@ -0,0 +1,19 @@
+use std::path::Path;
+
+pub struct Git {
+    repo: git2::Repository,
+}
+
+impl Git {
+    pub fn new<P: AsRef<Path>>(path: P) -> Self {
+        let repo = git2::Repository::discover(path).unwrap();
+        Self { repo }
+    }
+
+    pub fn is_uncommitted(&self, path: &Path) -> bool {
+        match self.repo.status_file(path) {
+            Ok(status) => matches!(status, git2::Status::WT_NEW),
+            _ => false,
+        }
+    }
```

Comment:
This is a little confusing, `WT_NEW` is "working tree new", once committed the status is `CURRENT`, so maybe it's better to just check `!matches!(status, git2::Status::CURRENT)`?